### DISCUSSION TITLE
Remap for "preview-page-down" and "preview-page-up" does not work | Help

### DISCUSSION THREAD

**Discussion by lsig** [ibhagwan/fzf-lua#1259] [Q&A] [answered]
So I have been trying to change the way I scroll previews using fzf-lua and for some reason I cannot get it to work. Other remaps have worked great but not this one. 

I started by looking at the docs and found the advanced config section. There I spotted this: 
![Screenshot 2024-06-13 at 20 20 24](https://github.com/ibhagwan/fzf-lua/assets/89947902/e0b96ff7-14e8-4a5e-9dcd-e6c0d79ca34e)

shift-down and shift-up allow me to scroll without using the mouse. So I changed my setup to this (I tried both just mapping ctrl-b and crtl-f to the preview functionality and doing that and setting the shift-up and down to false): 

fzf.setup({
      defaults = { formatter = { "path.filename_first" } },
      keymap = {
        fzf = {
          ["CTRL-Q"] = "select-all+accept",
          ["shift-up"] = false,
          ["shift-down"] = false,
          ["ctrl-b"] = "preview-page-up",
          ["ctrl-f"] = "preview-page-down",
        },
      },
      git = {
        status = {
          actions = {
            -- actions inherit from 'actions.files' and merge
            ["right"] = false,
            ["left"] = false,
            ["ctrl-l"] = { fn = actions.git_unstage, reload = true },
            ["ctrl-h"] = { fn = actions.git_stage, reload = true },
            ["ctrl-x"] = { fn = actions.git_reset, reload = true },
            ["ctrl-s"] = { fn = actions.git_stage_unstage, reload = true },
          },
        },
      },
    })

For some reason this does not work. Shift up and down continuous to scroll up and down and ctrl b and f do not do anything. Could I be looking at the wrong option or something?

I also tried setting:

builtin = {
          ["ctrl-f"] = "preview-page-down",
          ["ctrl-b"] = "preview-page-up",
    },

with no luck. But all the git remaps work like I charm. Any help would be appreciated. 

@ibhagwan Thanks for the amazing work and plugin 💯 .

**Comment by lsig** [comment, marked as answer]
I found the solution but maybe the documentation needs to be updated accordingly.
```lua
keymap = {
        builtin = {
          ["<C-f>"] = "preview-page-down",
          ["<C-b>"] = "preview-page-up",
        },
  },
```
This works but no other variation of "ctrl-f", "\<ctrl-f\>" or "\<Ctrl-f\>" does. Hope this helps someone down the line :)

**Comment by ibhagwan** [comment]
Ty for the kind words @lsig!

I need to work on making the binds a bit more intuitive, currently it requires understanding which previews is native (I.e native to fzf in the shell) which which one is a neovim buffer (aka builtin).

The mappings that are under `keymap.builtin` work for neovim buffers (where you have a neovim window as the preview.

Note that due to binds being overly personal I made it so the keymap tables override the defaults so you don’t have to set shift-up|down to false.

Now let’s try to isolate this issue, since you set this ok for native do the binds work with the bat previewer (which is native):
```lua
:FzfLua files previewer=bat
```

And now let’s try with the keymal variant:
```lua
:FzfLua files previewer=builtin keymap.builtin.ctrl-b=preview-page-up
```

**Comment by ibhagwan** [comment]
> I found the solution but maybe the documentation needs to be updated accordingly.

There’s a comment about this in the README above the tables but I agree there should be more clarity but somehow I always have better stuff to do than docs.

The wiki is open edit btw :)

**Reply by lsig** [reply]
Thanks for the quick response! 

Yeah the part that was confusing is that for the actions "ctrl-h" is enough but not for the builtin previewer (different format \<C-f\>).

I'll checkout the wiki and README to see if I can be of any help :)

**Reply by lsig** [reply]
wiki states it clearly:

When defining keymap.builtin binds you must use neovim-style keymaps (i.e. <C-c> for Control-C).

When defining keymap.fzf binds you must use fzf-style keymas (i.e. ctrl-c for Control-c), man fzf for all available fzf mappings.

**Reply by ibhagwan** [reply]
Can probably be consolidated for simplicity with a bit of work to translate neovim to fzf and vice versa.

The reason is that fzf binds are specified in the fzf shell command and neovim binds are set using `vim.keymap.set` directly on the fzf window and support greater key combos than fzf.

**Reply by lsig** [reply]
Yeah I am thinking about if updating the readme with what the wiki states is enought. Could submit a quick PR but its up to you, having it in the wiki is maybe enough.

**Reply by ibhagwan** [reply]
The Wiki should be alright at least until a proper bind refactor.

**Reply by lsig** [reply]
Sounds good
